### ISSUE TITLE
Upgrade JavaPoet dependency to fix version conflict with Dagger 2

### ISSUE THREAD

**Issue by joelpet** [pushtorefresh/storio#763]
In my Android project (that I can unfortunately not share) with StorIO 1.12.2 and Dagger 2.8, the Gradle build fails with the following exception:

```
Caused by: java.lang.NoSuchMethodError: com.squareup.javapoet.ClassName.topLevelClassName()Lcom/squareup/javapoet/ClassName;
        at dagger.internal.codegen.SourceFiles.siblingClassName(SourceFiles.java:209)
        at dagger.internal.codegen.SourceFiles.membersInjectorNameForType(SourceFiles.java:189)
        at dagger.internal.codegen.SourceFiles.generatedClassNameForBinding(SourceFiles.java:172)
        at dagger.internal.codegen.InjectBindingRegistry$BindingsCollection.tryToGenerateBinding(InjectBindingRegistry.java:107)
        at dagger.internal.codegen.InjectBindingRegistry$BindingsCollection.tryRegisterBinding(InjectBindingRegistry.java:99)
        at dagger.internal.codegen.InjectBindingRegistry.registerBinding(InjectBindingRegistry.java:210)
        at dagger.internal.codegen.InjectBindingRegistry.tryRegisterMembersInjectedType(InjectBindingRegistry.java:272)
        at dagger.internal.codegen.InjectBindingRegistry.tryRegisterMembersInjectedType(InjectBindingRegistry.java:251)
        at dagger.internal.codegen.InjectProcessingStep$1.visitVariableAsField(InjectProcessingStep.java:69)
        at dagger.internal.codegen.InjectProcessingStep$1.visitVariableAsField(InjectProcessingStep.java:59)
        at com.sun.tools.javac.code.Symbol$VarSymbol.accept(Symbol.java:1237)
        at dagger.internal.codegen.InjectProcessingStep.process(InjectProcessingStep.java:58)
        at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:329)
        at dagger.shaded.auto.common.BasicAnnotationProcessor.process(BasicAnnotationProcessor.java:182)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:794)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:705)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)
        at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)
        at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)
        at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)
        at com.sun.tools.javac.main.Main.compile(Main.java:523)
        ... 89 more
```

The problem seems to be that StorIO depends on an old version of JavaPoet (1.2), which is the version resolved by Gradle, while Dagger uses the `com.squareup.javapoet.ClassName.topLevelClassName()` method added in JavaPoet 1.4. I have resolved this in my project by forcing Gradle to resolve the JavaPoet dependency with the 1.8 version. 

However, I thought it would be nice if StorIO used the most recent JavaPoet version as well so I made the upgrade and ran the tests (`sh ci.sh`) just to see if they would still pass, which they did.

In summary, please upgrade to JavaPoet 1.4+.

**Comment by geralt-encore** [comment]
@joelpet Thank you for reporting the issue!
